### PR TITLE
NAS-110158 / None / Add EntityJobState enum

### PR DESCRIPTION
**Pull request by denysbutenko**
extends [PR#5315](https://github.com/truenas/webui/pull/5315), target branch: NAS-109874

### CODE REVIEW

File: src/app/pages/data-protection/components/data-protection-dashboard/data-protection-dashboard.component.ts [hunk 1]
```diff
@@ -809,55 +795,57 @@ export class DataProtectionDashboardComponent implements OnInit, OnDestroy {
...
+      //         this.ws.call('core.download', ['filesystem.get', [row.job.logs_path], filename]).subscribe(
+      //           (res) => {
+      //             const url = res[1];
+      //             const mimetype = 'text/plain';
+      //             this.storage.streamDownloadFile(this.http, url, filename, mimetype).subscribe(
+      //               (blob) => this.storage.downloadBlob(blob, filename),
+      //               (err) => new EntityUtils().handleWSError(this, err),
+      //             );
+      //           },
+      //           (err) => new EntityUtils().handleWSError(this, err),
```
Comment (undsoft) [inline src/app/pages/data-protection/components/data-protection-dashboard/data-protection-dashboard.component.ts:844]: What happened here?

Comment (denysbutenko) [reply src/app/pages/data-protection/components/data-protection-dashboard/data-protection-dashboard.component.ts:844]: I accidentally added uncompleted changes — marked PR as a draft until getting it fixed.

File: src/app/pages/common/entity/entity-job/entity-job.interface.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export enum EntityJobState {
+  pending = 'PENDING',
+  running = 'RUNNING',
+  hold = 'HOLD',
+  error = 'ERROR',
+  failed = 'FAILED',
+  aborted = 'ABORTED',
+  success = 'SUCCESS',
+  finished = 'FINISHED'
```
[the remark this answers is not in the document]
Comment (denysbutenko) [reply src/app/pages/common/entity/entity-job/entity-job.interface.ts:9]: I'm following the current style of other enums:
https://github.com/truenas/webui/blob/6a44afee3b76d42e264e117c64d97dded5dfb48a/src/app/pages/common/entity/entity-form/models/field-config.interface.ts#L3-L3
https://github.com/truenas/webui/blob/dc00d80b48fe828428985a94835b744c468c1291/src/app/pages/common/entity/entity-empty/entity-empty.component.ts#L7-L7

[the remark this answers is not in the document]
Comment (undsoft) [reply src/app/pages/common/entity/entity-job/entity-job.interface.ts:9]: Those are incorrect as well and would need to be fixed.

**Review by undsoft — APPROVED** [review]

**Pull request merged**